<a href="https://colab.research.google.com/github/Electromayonaise/Calculus-Notebooks/blob/main/Laboratorio2%7C%7C_Grupo_10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img src=http://www.acofi.edu.co/eiei2016/wp-content/uploads/2016/09/Logo-Universidad-ICESI.jpg width="300" height="200" align="center"> 

#**<big><big>Matemáticas Aplicadas I</big></big><br><small>Departamento de Matemáticas y Estadística / Facultad de Ingeniería</small><br><small> Laboratorio 2</small>**
Martín Gómez Caicedo.             Código:A00399958
<br>
Alejandro Quiñones C.             Código: A0037703
<br>
Mateo Rubio.                     Código: A00400104

In [ ]:
import numpy as np

In [ ]:
def Newton(f,df,x0,n):
  li=[x0]
  for i in range(n):
    x=li[-1]
    li.append(x-f(x)/df(x))
  return li

## Problema 1. 
Para construir la maqueta de un foso de lanzamiento de disco, modalidad de
atletismo, se usa una regla de madera flexible de 20 pulgadas de longitud que
se dobla en forma de arco circular como se indica la figura. Si la longitud de la
cuerda generada por el arco debe tener una longitud de 16 pulgada, determine
el radio del arco de circunferencia. 

Para esto necesitamos principalmente dos formulas, la primera seria 

$$ L= r * \Theta $$

Que nos sirve para hallar la longitud (*L*) de un arco en la circunferencia con su radio y angulo, ademas 

$$ K = 2 * r * sen(\Theta/2) $$

donde K es la longitud de la cuerda formada con el angulo *Teta* y el radio *r*

Para el ejercicio reemplazamos los valores que conocenmos y nos queda

$$ \begin{Bmatrix}
 & 20=\Theta * r\\ 
 & 16=2r*sen(\Theta /2)
\end{Bmatrix} $$

si despejamos a Teta en la primera ecuacion resulta en $20/r$ y al sustituir en la segunda ecuacion queda

$$ 16 = 2*r*sen(10/r) 
\\\ 8=r*sen(10/r) 
$$

Esta situacion no tiene solucion analitica, por lo que debemos buscar metodos alternativos para su solucion, que puede ser usar $f(r)=r*sen(10/r)$ y ver que pasa cuando $f(r)=8$, el metodo de biseccion o el metodo de Newton.

Para este ultimo tendriamos que derivar asi:

$$ \frac{\mathrm{d} }{\mathrm{d} r}[8]=\frac{\mathrm{d} }{\mathrm{d} r}[r*sen(10/r )]$$

$$f'(x)=sen(10/r)-\frac{10*cos(10/r)}{r}$$

In [ ]:
def a(x):return x*np.sin(10/x)-8
def da(x):return np.sin(10/x)-((10*np.cos(10/x))/x)

In [ ]:
Newton(a,da,8,6)

[8,
 8.735579942348492,
 8.83922259745267,
 8.84093062500307,
 8.840931076328516,
 8.840931076328548,
 8.840931076328548]

Por lo que podemos concluir que el radio es aproximadamente 8.841

##Problema 2. 
Demuestre que el método de Newton aplicado a la ecuación $x^{3}-a=0$ conduce a la fórmula $x{_{(n+1)}}=\frac{1}{3}\left ( 2x{_{n}}+\frac{a}{{x_{n}}^2}\right )$, usada para aproximar la raíz cúbica de $a$. Utilice el método de Newton para aproximar el valor de $\sqrt[3]{7}$ con una presición de cinco decimales.

Tenemos que: 

$f(x)= x^{3}-a \\\\
f`(x) = 3x^2
\\\\por\ lo \ tanto: \ x_{n+1}= x_{n}- \frac{x_{n}^3-a}{3x_{n}^2}
\\\\ x_{n+1}=  \frac{3x_{n}^3-(x_{n}^3-a)}{3x_{n}^2}
\\\\ x_{n+1}=  \frac{2x_{n}^3+a}{3x_{n}^2}
\\\\ x_{n+1}=  \frac{2x_{n}^3}{3x_{n}^2} + \frac{a}{3x_{n}^2}
\\\\ x_{n+1}=  \frac{2x_{n}}{3} + \frac{a}{3x_{n}^2}
\\\\ x_{n+1}= \frac{1}{3}(2x_{n} + \frac{a}{x_{n}^2}) $


Y así probamos algebraicamente que $x^{3}-a=0$ conduce a $ x_{n+1}= \frac{1}{3}(2x_{n} + \frac{a}{x_{n}^2}) $ algebraicamente. 

Por otro lado, para realizar una aproximación con presición de 5 decimales al valor de $\sqrt[3]{7}$ podemos usar la formula recién obtenida, por lo que: 

$x_{1}= \frac{1}{3}(2x_{n} + \frac{7}{x_{n}^2}) $ , si tomamos $x_0 = 2$ obtenemos: 

$x_{1}= \frac{1}{3}(2(2) + \frac{7}{2^2}) =  \frac{1}{3}(4 + \frac{7}{4})=\frac{23}{12} ≈ 1,916 $ , realizamos otra iteración 

$x_{2}= \frac{1}{3}(2(1,916) + \frac{7}{1,916^2}) = \frac{18215}{9522} ≈ 1,91293 $, aquí ya obtenemos una presición de 5 décimales (solo tomo 2 iteraciones) 




#PROBLEMA 3

Obtenemos r en función de x usando el teorema de pitágoras, después igualamos $\frac{\pi}{3}$ con $\frac{\pi x}{6} * (3r^2 + x^2)$

$$
(1-x)^2 + r^2 = 1\\
r^2 = 1 - (1-x)^2\\
r = \sqrt{1-(1-x)^2}\\\\
r = \sqrt{2x-x^2}\\\\
\therefore \:\frac{\pi}{3} = \frac{\pi x}{6} * (3*(\sqrt{2x-x^2})^2+x^2)\\\\
\rightarrow \:\frac{\pi}{3} = \frac{\pi x}{6} * (3*(2x-x^2)+x^2)\\\\
\rightarrow \:\frac{\pi}{3} = \frac{\pi x}{6} * (6x-3x^2+x^2)\\\\
\rightarrow \:\frac{\pi}{3} = \frac{\pi x}{6} * (6x-2x^2)\\\\
\rightarrow \:\frac{\pi}{3} = \frac{\pi x}{6} * 2(3x-x^2)\\\\
\rightarrow \:\frac{\pi}{3} = \frac{\pi x}{3} * (3x-x^2)\\\\
\rightarrow \:\pi = \pi x * (3x-x^2)\\\\
\rightarrow \:1 = x * (3x-x^2)\\\\
\rightarrow \:1 = 3x^2-x^3\\\\
\rightarrow \:1 = 3x^2-x^3\\\\
\rightarrow \:x^3-3x^2+1 = 0\\\\
$$

Haciendo uso del método de Newton obtenemos que las raíces son $x_{1} = -0.53209$, $x_{2} = 0.65270$, $x_{3} = 2.8794$.
Por lo tanto teniendo en cuenta el contexto en el cual estamos usando $x$ sabemos que $1-x$ debe ser mayor a 0 y $x$ no debe ser negativa, por lo tanto la raíz que cumple esto es $x_{2}$